### ISSUE TITLE
Fetching profile from a Waku backup is a bit broken

### ISSUE THREAD

**Issue by John-44** [status-im/status-desktop#10276]
I had to restore my account from seed phrase, and the following happened:

1) The "Fetching data..." screen was correctly displayed.  Notice that it's telling me that "1/1" of my profile records have been successfully fetched

![2023-04-12 18_06_25-Status Desktop](https://user-images.githubusercontent.com/22846356/231532678-7778b667-245f-4290-8974-c73561f6478a.png)

2) However after waiting for two minutes and then pressing the "Continue" button, I was taken to the 'Unable to fetch your profile' screen, which is the screen a user should see when their user profile data (username, avatar, etc...) has *not* been fetched.

![2023-04-12 18_08_16-Status Desktop](https://user-images.githubusercontent.com/22846356/231533074-50dbd38b-2a49-415f-8a2e-80628ca22862.png)

3) I then pressed the "Try to fetch profile again" button (even though I know this should only induce two minute delay to see if further data arrives from Waku).  The app appeared to crash, and was unresponsive for several minutes

4) Then just as I was about to force quite the app, it unfroze and took me back to the "Fetching data..." screen (it should have displayed this screen immediately after I pressed the "Try to fetch profile again" button on the proceeding screen).  Strangely, this screen now says that "9/10" of 'Community membership' information is downloaded, when the previous time this screen was displayed (in step 1) it said that "9/9" of 'Community membership' information was successfully downloaded,

![2023-04-12 18_11_07-Status Desktop](https://user-images.githubusercontent.com/22846356/231534545-351ad109-bef6-4bd8-95ca-79ba2dc99b78.png)

5) After waiting on the "Fetching data..." screen for around 5min I pressed the "Continue" button again, and was taken to the 'Unable to fetch your profile' screen again

![2023-04-12 18_08_16-Status Desktop](https://user-images.githubusercontent.com/22846356/231535098-2f82a6b0-a441-45c5-b70a-237938ec213e.png)

6) This time I pressed the "Create new profile with the same chatkey" button, and entered my chat name and chose an avatar image.  *Issue minor here: when I pressed the primary action button, the app froze for about a second before I was taken to the loading screen"

Good news: when I arrived back in Status, my communities were restored and it looks like my mutual contacts were restored :-)

** Summary of the issues found **

**Issue 1:** Restore of my username and avatar failed even though the "Fetching data..." screen told me that this information had been successfully restored.

**Issue 2:** App froze for several minutes after pressing the "Try to fetch profile again" button on the 'Unable to fetch your profile' screen,

**Issue 3:** The second time I viewed the "Fetching data..." screen (after pressing the "Try to fetch profile again" button), it showed that there were 10 pieces of community membership information (even though the last time I was on this screen in the same flow it said there were only 9 pieces of community membership information).  This 10th piece of community membership information never downloaded, even after waiting for more than 5 minutes.

**Issue 4.** Even though my mutual contacts had been restored, the secondary nav bar in the messages section was empty.  I know that the chat history for 1on1 and ad-hoc group chats cannot be restored due to 'perfect forward secrecy', however at a minimum I should be able to see the ad-hoc group chats that I'm part of (even though there would be no history in them), because without this I have no way of knowing the ad-hoc group chats I'm in.  It would also be good for the 1on1 chats to be restored (again, even though the history in these chats can't be restored).   

So I think we should:
- Add a list of the ad-hoc group chats and 1on1 chats the user is in, and the *recency order* of these chats to the data we back up to Waku.
- When the user restores from a backup to waku, we should use this data to repopulate the list of 1on1 and ad-hoc group chats the user is in in the 'Messages' section of the app. 

**Possible minor improvement idea:** Because I haven't been using a keycard, on the "Fetching data..." screen the "Keycards" spinner was always spinning
![2023-04-12 18_30_48-2023-04-12 18_06_25-Status Desktop png ‎- Photos](https://user-images.githubusercontent.com/22846356/231537630-5003bc97-33fd-4896-aa84-8431e251a2c1.png)
If a user isn't doesn't have any data to back up to waku in any category, how about if we backup to waku a message which tells us that 'the user has no information to backup to waku in this category'.  In my particular case that's described above, such a message could be used to show "0/0" and a green bar against "Keycards".  

-----------------------------

There are bunch of issues in this issue, they should probably be split up into different issues so they can be worked on separately.  Sorry for not doing this, I just wanted to get everything I had just experienced written down quickly, and I've got a bunch of other things waiting on me.

**Comment by jrainville** [comment]
> I have lost some mutual contacts, instead of them being Contacts, they are sitting under Settings / Messaging / Contacts / Pending Requests screen.

This is known and being fixed by @MishkaRogachev

**Comment by MishkaRogachev** [comment]
> > I have lost some mutual contacts, instead of them being Contacts, they are sitting under Settings / Messaging / Contacts / Pending Requests screen.
> 
> This is known and being fixed by @MishkaRogachev

This one should be fixed by this PR: https://github.com/status-im/status-desktop/pull/10336

**Comment by saledjenic** [comment]
Recovering wallet account and keycards from another device or waku should be fixed in this PR: https://github.com/status-im/status-desktop/pull/10724

Important: 
- before testing this make sure that your data are backed up (old backed up data to waku doesn't fit the new form we're using)
- restoring feature in terms of wallet accounts + keycards work well if user is Status regular user (means profile keypair is not migrated to a keycard, this case will be handled in a new PR)
